e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts

```diff
@@ -0,0 +1,195 @@
+import axios from "axios";
+import { deploySc, sendTransaction, fundAddress, issueEsdt, transferEsdt, IssueEsdtArgs, TransferEsdtArgs, DeployScArgs } from "./chain.simulator.operations";
+import * as fs from "node:fs";
+import { EventProcessor } from "../../src/event.processor";
+import { EventProcessorOptions } from "../../src/types/event.processor.options";
+
+const CHAIN_SIMULATOR_URL = 'http://localhost:8085';
+const ELASTIC_SEARCH_URL = 'http://localhost:9200';
+const ALICE_ADDRESS = 'erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th';
+const BOB_ADDRESS = 'erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx';
+const CAROL_ADDRESS = 'erd1k2s324ww2g0yj38qn2ch2jwctdy8mnfxep94q9arncc6xecg3xaq6mjse8';
+const ONE_EGLD = '1000000000000000000';
+const VERBOSE_LOGS = false;
+
+describe('EventProcessor e2e tests with chain simulator', () => {
+  let eventProcessor: EventProcessor;
+
+  beforeAll(async () => {
+    try {
+      const response = await axios.get(`${CHAIN_SIMULATOR_URL}/simulator/observers`);
+
+      let numRetries = 0;
+      while (true) {
+        if (response.status === 200) {
+          await axios.post(`${CHAIN_SIMULATOR_URL}/simulator/generate-blocks-until-epoch-reached/2`, {});
+          break;
+        }
+
+        numRetries += 1;
+        if (numRetries > 50) {
+          fail("Chain simulator not started!");
+        }
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  });
+
+  beforeEach(() => {
+    eventProcessor = new EventProcessor();
+    jest.clearAllMocks();
+  });
+
+  it('should deploy a contract and receive all the ping pong events', async () => {
+    const contractCodeRaw = fs.readFileSync('./e2e-tests/chain-simulator-e2e/contracts/ping-pong/ping-pong-egld.wasm');
+    /*
+    fn init(
+        &self,
+        ping_amount: &BigUint,
+    ) {
+     */
+    const contractArgs = [
+      '0de0b6b3a7640000', // 1 egld
+    ];
+    await fundAddress(CHAIN_SIMULATOR_URL, ALICE_ADDRESS);
+    const scAddress = await deploySc(new DeployScArgs({
+      chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+      deployer: ALICE_ADDRESS,
+      contractCodeRaw: contractCodeRaw,
+      hexArguments: contractArgs,
+    }));
+    logMessage(`Deployed ping pong SC. Address: ${scAddress}`);
+
+    const numPingPongs = 20;
+
+    for (let i = 0; i < numPingPongs; i++) {
+      const pingTxHash = await sendTransaction({
```

Comment:
* same for the sendTransaction, maybe it's more robust to create a function for sendTransaction

Reply:
well, i call it with different arguments each time. I don't see a benefit extracting a function that just calls a function